contracts/TokenVesting.sol

```diff
@@ -0,0 +1,140 @@
+        uint256 redeemableUnlockedTokens = getRedeemableUnlockedTokens(msg.sender);
+        require(redeemableUnlockedTokens > 0, "You cannot claim any tokens at the moment"); // Placed require statement here so that getRedeemableUnlockedTokens(msg.sender) doesn't have to be computed twice
+        redeemedInvestorTokens[msg.sender] += redeemableUnlockedTokens;
+        SafeERC20.safeTransfer(IERC20(solace), msg.sender, redeemableUnlockedTokens);
+    }
+
+    /**
+     * @notice Calculates the amount of unlocked SOLACE tokens an investor can claim.
+     * @param investor Investor address.
+     * @return redeemableUnlockedAmount The amount of unlocked tokens an investor can claim from the smart contract.
+     */
+    function getRedeemableUnlockedTokens(address investor) public view override returns (uint256 redeemableUnlockedAmount) {
+        uint256 timestamp = block.timestamp;
+        uint256 redeemableUnlockedAmount;
+        if(timestamp <= cliff) {
+            return 0;
+        } else if(timestamp <= vestingEnd) {
+            uint256 totalUnlockedAmount = ( totalInvestorTokens[investor] * (timestamp - cliff) / (vestingEnd - cliff) );
+            redeemableUnlockedAmount = totalUnlockedAmount - redeemedInvestorTokens[investor];
+            return redeemableUnlockedAmount;
+        } else {
+            redeemableUnlockedAmount = totalInvestorTokens[investor] - redeemedInvestorTokens[investor];
+            return redeemableUnlockedAmount;
+        }
+    }
+
+    /***************************************
+    GOVERNANCE FUNCTIONS
+    ***************************************/
+
+    /**
+     * @notice Rescues excess [**SOLACE**](./SOLACE).
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @dev Trusting governance to perform accurate accounting off-chain and ensure there is sufficient SOLACE in contract to make payouts as dictated in totalInvestorTokens mapping.
+     * @param amount Amount to send.
+     * @param recipient Address to send rescued SOLACE tokens to.
+     */
+    function rescueSOLACEtokens(uint256 amount, address recipient) external override onlyGovernance {
+        require(recipient != address(0x0), "zero address recipient");
+        SafeERC20.safeTransfer(IERC20(solace), recipient, amount);
+    }
+
+    /**
+     * @notice Sets the total SOLACE token amounts that investors are eligible for.
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @dev Trusting governance to perform accurate accounting off-chain and ensure there is sufficient SOLACE in contract to make payouts as dictated in totalInvestorTokens mapping.
+     * @param investors Array of investors to set.
+     * @param totalTokenAmounts Array of token amounts to set.
+     */
+    function setTotalInvestorTokens(address[] calldata investors, uint256[] calldata totalTokenAmounts) external override onlyGovernance {
```

Comment:
Again red flag. We shouldn't change investor token balances during function call. In constructor, we can make such assignments.

Reply:
the list will likely change. when its finalized we can lock governance